python/core/additions/qgsfunction.py

```diff
@@ -48,11 +48,24 @@ def myfunc(values, *args):
 
     Eval errors can be raised using parent.setEvalErrorString("Error message")
 
-    :param function:
-    :param arg_count:
-    :param group:
-    :param usesgeometry:
-    :param handlesnull: Needs to be set to True if this function does not always return NULL if any parameter is NULL. Default False.
+    :param function: Function to be registered.
+    :param arg_count: Defines the number of arguments. With args="auto" the number of
+                    arguments will automatically be extracted from the signature. With
+                    args=-1, any number of arguments are accepted. Note that two
+                    arguments, feature and parent, are always automatically passed.
```

Comment:
Are these also counted?

Reply:
I think these two should be listed as "parameters" passed to the python function and _not_ as arguments (passed to the expression function) that would keep taxonomy clearer.

Reply:
> Are these also counted?

It's ... complicated and seems quite buggy (or at least full of unexpected behaviour). And sadly there is little to no inline documentation talking about the concepts and what the code *should* do. Let's handle that in a separate issue.

The "parameters" vs. "arguments" is confusing me but sounds like a very good thing to be very correct about. We would need some clarification in the text (in `register_function`'s docstring) just about the changed lines as well, right?

Reply:
> > Are these also counted?
> 
> It's ... complicated and seems quite buggy (or at least full of unexpected behaviour).

How so?

> 
> The "parameters" vs. "arguments" is confusing me but sounds like a very good thing to be very correct about. We would need some clarification in the text (in `register_function`'s docstring) just about the changed lines as well, right?

That was a proposal, not established taxonomy.

Example:

A new function `add_one(value)` is defined. Arg count is `1` because there is only one argument in the created function.

The python "backing function" still receives a feature and parent. These should IMHO not be mentioned in any way that they could be mistaken for an argument of the expression function.

Does that clarify?